furnace.c

```diff
@@ -58,6 +61,9 @@ typedef struct {
 
   pilot_context_t       pilot;
   uint8_t pwm_level;
+
+  uint8_t         pulse_count;
+  absolute_time_t magnetron_deadline;
 } furnace_context_t;
 
 #include "pwm.c"
```

Comment:
Instead of inserting several `#if CONFIG_MAGNETRON`, we might just include `magnetron.c` and guard here. Including C source allows for `static` functions and usually results in better compile times.